pubnub/event_engine/effects.py

```diff
@@ -0,0 +1,431 @@
+import asyncio
+import logging
+import math
+
+from typing import Optional, Union
+from pubnub.endpoints.presence.heartbeat import Heartbeat
+from pubnub.endpoints.presence.leave import Leave
+from pubnub.endpoints.pubsub.subscribe import Subscribe
+from pubnub.enums import PNReconnectionPolicy
+from pubnub.exceptions import PubNubException
+from pubnub.features import feature_enabled
+from pubnub.models.server.subscribe import SubscribeMessage
+from pubnub.pubnub import PubNub
+from pubnub.event_engine.models import events, invocations
+from pubnub.models.consumer.common import PNStatus
+from pubnub.workers import BaseMessageWorker
+
+
+class Effect:
+    pubnub: PubNub = None
+    event_engine = None
+    invocation: Union[invocations.PNManageableInvocation, invocations.PNCancelInvocation]
+    stop_event = None
+    logger: logging.Logger
+    task: asyncio.Task
```

Comment:
Nitpicking:
It is easy to make a mistake when looking at the default values. 
```suggestion
    pubnub: PubNub = None
    event_engine: StateMachine = None
    stop_event: PNEvent = None
    invocation: Union[invocations.PNManageableInvocation, invocations.PNCancelInvocation]
    logger: logging.Logger
    task: asyncio.Task
```